extensions/amp-sidebar/1.0/amp-sidebar.js

```diff
@@ -62,12 +76,21 @@ class AmpSidebar extends BaseElement {
     super.afterOpen();
     const sidebar = this.element.shadowRoot.querySelector('[part=sidebar]');
     this.setAsContainer?.(sidebar);
+
+    this.history_
+      .push(() => this.api().close())
+      .then((historyId) => (this.historyId_ = historyId));
   }
 
   /** @override */
   afterClose() {
     super.afterClose();
     this.removeAsContainer?.();
+
+    if (this.historyId_ != -1) {
```

Comment:
The goal is that hitting the back/forward will trigger Sidebar opening/closing?

Reply:
> The goal is that hitting the back/forward will trigger Sidebar opening/closing?

Yes

> Do you think the Preact component would be interested in behavior like this as well?

Most likely, yes, but that is a larger project and not a requirement for backwards compat with the AMP layer :/ cc @nainar as FYI to add this to the roadmap

Reply:
Thanks for the explanation / context. I didn't realize sidebar did this!